code/script.js

```diff
@@ -232,94 +235,175 @@ const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
   // What else should happen when we start the game?
-}
+  generateBoard();
+  setSecret();
+  // console.log(secret)
+};
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
 
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  const value = questions.options[questions.selectedIndex].value
 
   currentQuestion = {
     category: category,
-    // value: value
+    value: value,
   }
+  checkQuestion()
 }
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
   const { category, value } = currentQuestion
+  // console.log(category)
+  // console.log(value)
 
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
   if (category === 'hair' || category === 'eyes') {
+      if (secret.hair === value || secret.eyes === value) {
+          filterCharacters(keep)
+      } else {
+          filterCharacters()
+      }
 
   } else if (category === 'accessories' || category === 'other') {
-
+      if (secret.accessories[0] === value || secret.other[0] === value) {
```

Comment:
I would also think in terms of scalability here. Right now, hard-coding the index of the accessories array isn't a problem because there is only one index anyway. But if you wanted to add more items, you'd need to use a variable instead of an index number. 

You could do something similar to line 289, where you used `person[category]` instead of a hard-coded index.